modules/candidate_parameters/ajax/formHandler.php

```diff
@@ -477,10 +478,11 @@ function editConsentStatusFields($db, $user)
                           requires only the date of consent.');
                     return;
                 }
-            } else { // If no status stays no or record existed as NULL,
-                     // consent date and empty withdrawal date still required
+            } else { // If no status stays no or record existed as NULL
+                    // consent date and empty withdrawal date still required
+                    // old withdrawal may be filled in
                 if (($oldStatus === null || $oldStatus === 'no') && !empty($date)
-                    && empty($withdrawal)
+                    && (empty($withdrawal) || !empty($oldWithdrawal))
```

Comment:
i think this allows you to delete an existing withdrawn date without having changed the status

Reply:
@zaliqarosli do you think they should be able to edit the withdrawal date, or should they not be able to make any changes at all ?

Reply:
i think we should allow them to edit date

Reply:
I dont think theres a better way of doing this simply from the oldstatus info since after 2 reloads we cant possibly know if it was a `yes -> no` change, `no -> no` or `null -> no` 2 reloads ago.

I think there are 2 choices.

1. digging in history to find if the status was ever set to `yes` and requiring the withdrawal date
2. allowing them to remove the withdrawal date after reloading the page...

I think given the amount of changes required for something that simple, option 2 is acceptable, with a note in the code saying this behaviour is expected although  not idea

Reply:
@ridz1208 I had originally done this with option 1 implemented. Should I just revert to that ? Either option is okay with me.

Reply:
I'd have to see how you were doing it, which commit is it ?

Reply:
The first one